### ISSUE TITLE
How to remove a part detection from hyperpose

### ISSUE THREAD

**Issue by sungs0711** [tensorlayer/hyperpose#298]
I've been trying to remove head part detection in real-time hyperpose through webcam. however im so short for c++ language and want to know how to deal with it. would you some one help me to detect with only excluding above body by code?

**Comment by ganler** [comment]
Hi @sungs0711 

## TL;DR

Add the following codes [here](https://github.com/tensorlayer/hyperpose/blob/3e71082c0424ae65d6137f54a75838a20766286d/examples/operator_api_imshow_paf.example.cpp#L76).

```c++
for (auto&& pose : poses) {
    pose.parts[0].has_value = false;
    pose.parts[14].has_value = false;
    pose.parts[15].has_value = false;
    pose.parts[16].has_value = false;
    pose.parts[17].has_value = false;
}
```

Then recompile it. (BTW, I did not actually try this out, so feel free to ask me again if this cannot work)

Feel free to ask us for further questions and thx for using our library.

---

## A more detailed guidance

1. Let's look at the `.process` method whose return type is `vector<human_t>`: https://github.com/tensorlayer/hyperpose/blob/master/examples/operator_api_batched_images_paf.example.cpp#L73
2. The class/structure `human_t` stands for COCO based keypoint for one single person.
3. Let's look at the COCO pose model.

Let's look at the key points first:

keypoint id: 14~17 along with 0 is the head.

![image](https://user-images.githubusercontent.com/38074777/90090990-fa552a00-dd57-11ea-9708-81156e094b45.png)

Finally, all you need to do is to set the `human_t::part::has_value` to `false` for head parts before we apply `draw_human` function.

**Comment by sungs0711** [comment]
i really appreciate you for your help, considerate and it works very nicely :). i feel like to make really nice project in univ with your big help. wish you get luck!.

Plus, if you don't mind to help me one more! then actually i have tried it all to increase FPS for real time detection and i thought removing some lines and points would support it.

However, unfortunately, FPS has been increased a little. furthermore, i have not used GPU like using only spoon to eat noodle! hehe..... so, anyway i tried to use gpu for hyperpose but it failed also. i think big problem is that i can handle not c++ but only python. thus, i want to ask you one more for helping us to deal with hyperpose in gpu environment. 

thank you for your kindness mercy love and genious hehe:)

**Comment by ganler** [comment]
@sungs0711  So did you mean that you had trouble in building our C++ project?

The Python codes also has the functionality to do inference, though it may be much slower than the C++ implementation.

We have documentation on how to build the C++ code. (I knew it is hard for guys who is not familiar with C++, especially we depend on the NVidia TensorRT library is totally user-unfriendly from my opinion)

If you are in a hurry, you can try our documentation out. Note that before you execute the commands [here](https://github.com/tensorlayer/hyperpose/pull/282), you need to install CUDA 10 and TensorRT 7. ([This guidance](https://ddkang.github.io/2020/01/02/installing-tensorrt.html) may work for you if you are using a **Linux** system.)
If you still meet any troubles in building, please tell us with some useful information(e.g., your OS version, the cmds you typed, and so on)

If you are not in a hurry, you can wait for our docker support(WIP).

Good luck.

**Comment by ganler** [comment]
Closed. Feel free to reopen it if new troubles occur. :-)
